pandas/core/indexes/multi.py

```diff
@@ -2085,8 +2085,22 @@ def sortlevel(self, level=0, ascending=True, sort_remaining=True):
             shape = list(self.levshape)
 
             # partition codes and shape
-            primary = tuple(codes.pop(lev - i) for i, lev in enumerate(level))
-            primshp = tuple(shape.pop(lev - i) for i, lev in enumerate(level))
+            primary = []
+            primshp = []
+
+            # Add the required codes and shape
```

Comment:
write both of these as a list comprehensiion (as was original, just w/o the pop)